In [1]:
!pip install selenium
!pip install haversine
!pip install folium
!pip install googlemaps

  Using cached urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.25.11
    Uninstalling urllib3-1.25.11:
      Successfully uninstalled urllib3-1.25.11
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
requests 2.23.0 requires urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1, but you have urllib3 1.26.7 which is incompatible.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.
  Using cached urllib3-1.25.11-py2.py3-none-any.whl (127 kB)
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.26.7
    Uninstalling urllib3-1.26.7:
      Successfully uninstalled urllib3-1.26.7
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.

In [1]:
import googlemaps
import folium
import haversine

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings(action="ignore") #경고 표시 제거는 일시적으로만 하자... 중요한 내용일수도

import matplotlib.font_manager as fm #한글 안깨지게
# font_name=fm.FontProperties(fname="C:/Windows/Fonts/malgun.ttf").get_name()
# plt.rc("font", family=font_name)

import matplotlib as mpl # - 표시 정상적으로 되도록
mpl.rcParams["axes.unicode_minus"]=False

import urllib.request as req
from urllib.request import urlopen
from urllib.error import HTTPError, URLError
import urllib.parse
from bs4 import BeautifulSoup
import requests

from selenium import webdriver
import time

In [ ]:
aa = pd.read_excel("data/망고플레이트 리뷰 종합.xlsx")
aa

Unnamed: 0       이름       평점                       주소            주력 메뉴  \
0          3869.0     청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n   
1          3865.0     청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n   
2          3867.0     청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n   
3          3868.0     청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n   
4          3863.0     청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n   
...           ...      ...      ...                      ...              ...   
36203         NaN  키미노돼지국밥        -  서울특별시 중랑구 중랑천로10길 92 1F  \n탕 / 찌개 / 전골\n   
36204         NaN    조개나루터        -     서울특별시 중랑구 중랑천로12길 59       \n해산물 요리\n   
36205         NaN    조개나루터        -     서울특별시 중랑구 중랑천로12길 59       \n해산물 요리\n   
36206         NaN    제주돈선생        -     서울특별시 중랑구 중랑천로12길 61        \n고기 요리\n   
36207         NaN    제주돈선생        -     서울특별시 중랑구 중랑천로12길 61        \n고기 요리\n   

          가격대                                          망고플레이트 주소  \
0      4만원 이상  https://www.mangoplate.com//restaurants/YyRGx0...   
1      4만원 이상  https://www.mangoplate.com//restaurants/YyRGx0...   
2      4만원 이상  https://www.mangoplate.com//restaurants/YyRGx0...   
3      4만원 이상  https://www.mangoplate.com//restaurants/YyRGx0...   
4      4만원 이상  https://www.mangoplate.com//restaurants/YyRGx0...   
...       ...                                                ...   
36203   만원 미만  https://www.mangoplate.com//restaurants/OGDe64...   
36204     NaN  https://www.mangoplate.com//restaurants/MgyUt1...   
36205     NaN  https://www.mangoplate.com//restaurants/MgyUt1...   
36206  만원-2만원  https://www.mangoplate.com//restaurants/JtAAHT...   
36207  만원-2만원  https://www.mangoplate.com//restaurants/JtAAHT...   

                                                    리뷰내용  
0      \n          가로수길에 있는 오랜 참치맛집 e-청우참치위치를 옮기다!!원래...  
1      \n          가성비 너무 좋고, 정말 양질의 참치를 양껏 즐길 수 있어서 ...  
2      \n          가성비 좋은 한국식 참치횟집. 선도가 꽤 좋다. 카운터석도 있...  
3        \n          그저 박수만. 참치땡길때 여기만한 곳이 없다.\n          
4           \n          맛있다.. 느끼해서 많이 못먹는게 함정.\n          
...                                                  ...  
36203  \n          뜨끈한 돼지국밥 생각 날 때 자주 찾는 곳.멀리서도 먹으러 찾...  
36204  \n          11월말에 처음 가보고나서 12~1월두달동안 벌써 8번이나 다...  
36205  \n          가리비크기 미친거아닌가요.......내일 또갈래요..........  
36206  \n          굳이 없는 식당까지 등록해가며 쓰지 않는데, 고기도 맛있고 사...  
36207  \n          완전 맛있네오̊̈!̆̈  전 김치찌개 너무 맛있었어요 죽이 별...  

[36208 rows x 8 columns]

In [ ]:
aa=aa.drop(["Unnamed: 0"], axis=1)
aa

이름       평점                       주소            주력 메뉴     가격대  \
0         청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n  4만원 이상   
1         청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n  4만원 이상   
2         청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n  4만원 이상   
3         청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n  4만원 이상   
4         청우참치  \n4.5\n       서울특별시 강남구 강남대로 632       \n회 / 스시\n  4만원 이상   
...        ...      ...                      ...              ...     ...   
36203  키미노돼지국밥        -  서울특별시 중랑구 중랑천로10길 92 1F  \n탕 / 찌개 / 전골\n   만원 미만   
36204    조개나루터        -     서울특별시 중랑구 중랑천로12길 59       \n해산물 요리\n     NaN   
36205    조개나루터        -     서울특별시 중랑구 중랑천로12길 59       \n해산물 요리\n     NaN   
36206    제주돈선생        -     서울특별시 중랑구 중랑천로12길 61        \n고기 요리\n  만원-2만원   
36207    제주돈선생        -     서울특별시 중랑구 중랑천로12길 61        \n고기 요리\n  만원-2만원   

                                               망고플레이트 주소  \
0      https://www.mangoplate.com//restaurants/YyRGx0...   
1      https://www.mangoplate.com//restaurants/YyRGx0...   
2      https://www.mangoplate.com//restaurants/YyRGx0...   
3      https://www.mangoplate.com//restaurants/YyRGx0...   
4      https://www.mangoplate.com//restaurants/YyRGx0...   
...                                                  ...   
36203  https://www.mangoplate.com//restaurants/OGDe64...   
36204  https://www.mangoplate.com//restaurants/MgyUt1...   
36205  https://www.mangoplate.com//restaurants/MgyUt1...   
36206  https://www.mangoplate.com//restaurants/JtAAHT...   
36207  https://www.mangoplate.com//restaurants/JtAAHT...   

                                                    리뷰내용  
0      \n          가로수길에 있는 오랜 참치맛집 e-청우참치위치를 옮기다!!원래...  
1      \n          가성비 너무 좋고, 정말 양질의 참치를 양껏 즐길 수 있어서 ...  
2      \n          가성비 좋은 한국식 참치횟집. 선도가 꽤 좋다. 카운터석도 있...  
3        \n          그저 박수만. 참치땡길때 여기만한 곳이 없다.\n          
4           \n          맛있다.. 느끼해서 많이 못먹는게 함정.\n          
...                                                  ...  
36203  \n          뜨끈한 돼지국밥 생각 날 때 자주 찾는 곳.멀리서도 먹으러 찾...  
36204  \n          11월말에 처음 가보고나서 12~1월두달동안 벌써 8번이나 다...  
36205  \n          가리비크기 미친거아닌가요.......내일 또갈래요..........  
36206  \n          굳이 없는 식당까지 등록해가며 쓰지 않는데, 고기도 맛있고 사...  
36207  \n          완전 맛있네오̊̈!̆̈  전 김치찌개 너무 맛있었어요 죽이 별...  

[36208 rows x 7 columns]

In [ ]:
aa.to_csv("data/망고플레이트 리뷰 종합.csv", sep=",")

In [ ]:
import googlemaps

In [ ]:
review_list = pd.read_csv("data/망고플레이트 리뷰 종합.csv", sep=",", index_col=0)
address=review_list["주소"].unique()
address

array(['서울특별시 강남구 강남대로 632', '서울특별시 강남구 강남대로110길 26 1F',
       '서울특별시 강남구 강남대로110길 62', ..., '서울특별시 중랑구 중랑천로10길 92 1F',
       '서울특별시 중랑구 중랑천로12길 59', '서울특별시 중랑구 중랑천로12길 61'], dtype=object)

In [ ]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')
geocode_result = gmaps.geocode(address[0])
geometry=geocode_result[0].get("geometry")
location=geometry.get("location")
lat=location.get("lat")
lng=location.get("lng")
print(geometry)

{'location': {'lat': 37.5182046, 'lng': 127.019356}, 'location_type': 'ROOFTOP', 'viewport': {'northeast': {'lat': 37.5195535802915, 'lng': 127.0207049802915}, 'southwest': {'lat': 37.5168556197085, 'lng': 127.0180070197085}}}


In [ ]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

for i in range(3):
    geocode_result = gmaps.geocode(address[i])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lat=location.get("lat")
    lng=location.get("lng")
    print("({}, {})".format(lng,lat))

(127.019356, 37.5182046)
(127.0269451, 37.5035911)
(127.0294865, 37.5033264)


In [ ]:
len(address)

3732

In [6]:
from haversine import haversine
from sympy import Symbol, solve

In [ ]:
haversine((37.5182046, 127.019356), ((37.5182046+37.5035911)/2, (127.019356+127.0269451)/2))

0.8787063094144727

In [ ]:
haversine((37.5035911, 127.0269451), ((37.5182046+37.5035911)/2, (127.019356+127.0269451)/2))

0.8787187891266018

In [ ]:
def middle_point():
    a=float(input("위도 1: "))
    b=float(input("경도 1: "))
    c=float(input("위도 2: "))
    d=float(input("경도 2: "))
    for x in np.arange(126.734086, 127.269311, 0.000001):
        for y in np.arange(37.413294, 37.715133, 0.000001):
            if haversine((a,b), (x,y)) == haversine((c,d), (x,y)):
                print(haversine((a,b),(x,y)))
                print(x,y)

In [ ]:
middle_point()

위도 1: 37.5182046
경도 1: 127.019356
위도 2: 37.5035911
경도 2: 127.0269451


KeyboardInterrupt: 

### 서울 경도 범위 (126.734086, 127.269311)
### 서울 위도 범위 (37.413294, 37.715133)

In [ ]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

latitude=[]
longitude=[]
coordinate=[]

for i in range(3):
    geocode_result = gmaps.geocode(address[np.random.randint(0, len(address))])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lng=location.get("lng")
    lat=location.get("lat")
    latitude.append(lat)
    longitude.append(lng)
    coordinate.append((lat, lng))
    print("({}, {})".format(lat,lng))

NameError: ignored

In [ ]:
for i in range(3):
    print(haversine(coordinate[i], (sum(latitude)/3, sum(longitude)/3)))

NameError: ignored

In [ ]:
print(sum(longitude)/3)
print(sum(latitude)/3)

다각형의 최소/최대 4점을 알면 중심점을 구할 수 있습니다.

x1 : 다각형의 x 좌표 중 가장 작은 값  
y1 : 다각형의 y 좌표 중 가장 작은 값  
x2 : 다각형의 x 좌표 중 가장 큰 값  
y2 : 다각형의 y 좌표 중 가장 큰 값  

center.x = x1 + ((x2 - x1) / 2);  
center.y = y1 + ((y2 - y1) / 2);  

In [ ]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

latitude=[]
longitude=[]
coordinate=[]

for i in range(5):
    geocode_result = gmaps.geocode(address[np.random.randint(0, len(address))])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lng=location.get("lng")
    lat=location.get("lat")
    latitude.append(lat)
    longitude.append(lng)
    coordinate.append((lat, lng))
    print("({}, {})".format(lat,lng))

In [28]:
x1=min(latitude)
x2=max(latitude)
y1=min(longitude)
y2=max(longitude)

center_x=x1+((x2-x1)/2)
center_y=y1+((y2-y1)/2)

for i in range(len(coordinate)):
    print(haversine((center_x, center_y), coordinate[i]))

TypeError: ignored

In [5]:
import folium

In [ ]:
x1=min(latitude)
x2=max(latitude)
y1=min(longitude)
y2=max(longitude)

center_x=x1+((x2-x1)/2)
center_y=y1+((y2-y1)/2)

map=folium.Map(location=[sum(latitude)/len(latitude),sum(longitude)/len(longitude)], zoom_start=11)
for i in range(len(coordinate)):
    folium.Marker([latitude[i], longitude[i]], popup="center").add_to(map)
folium.Marker([center_x, center_y], popup="center", icon=folium.Icon(color="red")).add_to(map)
map

In [ ]:
print(latitude)
print(longitude)
print(coordinate)


In [152]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [14]:
review_list = pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/망고플레이트 리뷰 종합.csv", sep=",", index_col=0)
address=review_list["주소"].unique()
address

array(['서울특별시 강남구 강남대로 632', '서울특별시 강남구 강남대로110길 26 1F',
       '서울특별시 강남구 강남대로110길 62', ..., '서울특별시 중랑구 중랑천로10길 92 1F',
       '서울특별시 중랑구 중랑천로12길 59', '서울특별시 중랑구 중랑천로12길 61'], dtype=object)

In [15]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

latitude=[]
longitude=[]
coordinate=[]

for i in range(3):
    geocode_result = gmaps.geocode(address[np.random.randint(0, len(address))])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lng=location.get("lng")
    lat=location.get("lat")
    latitude.append(lat)
    longitude.append(lng)
    coordinate.append((lat, lng))
    print("({}, {})".format(lat,lng))

(37.5317635, 126.9726135)
(37.5062698, 126.9616401)
(37.6064241, 127.0982814)


In [10]:
def get_norm(point1, point2):
            return sqrt((point1[0] - point2[0]) * (point1[0] - point2[0]) + \
                        (point1[1] - point2[1]) * (point1[1] - point2[1]))

In [13]:
from math import sqrt 

get_norm(coordinate[0], coordinate[1])

0.07897211582964281

In [ ]:
for x in np.arange(min(latitude), max(latitude), 0.0000001):
    for y in np.arange(min(longitude), max(longitude), 0.0000001):
        if 

In [11]:
!pip install scipy

In [29]:
from haversine import haversine

In [19]:
from scipy.stats.mstats import gmean

gm=gmean([coordinate[0], coordinate[1], coordinate[2]])

In [31]:
gm_point=(gm[0], gm[1])
print(gm_point)
d1=haversine(gm_point, coordinate[0])
d2=haversine(gm_point, coordinate[1])
d3=haversine(gm_point, coordinate[2])
print(d1, d2, d3)

(37.548128425482915, 127.01082987621912)
3.829495010217831 6.362433829912235 10.070368153527522


In [32]:
center_point=(sum(latitude)/len(latitude), sum(longitude)/len(longitude))
for i in range(len(coordinate)):
    print(haversine(center_point, coordinate[i]))

3.831938494180394
6.365298332820028
10.067626553406361


In [46]:
def center_pointer(num_points):
    import googlemaps
    import folium
    from haversine import haversine

    gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')
    latitude=[]
    longitude=[]
    coordinate=[]

    for i in range(num_points):
        address = input("도로명주소를 입력하세요: ")
        geocode_result = gmaps.geocode(address)
        geometry=geocode_result[0].get("geometry")
        location=geometry.get("location")
        lng=location.get("lng")
        lat=location.get("lat")
        latitude.append(lat)
        longitude.append(lng)
        coordinate.append((lat, lng))

    center_point=(sum(latitude)/len(latitude), sum(longitude)/len(longitude))
    map=folium.Map(location=[center_point[0], center_point[1]], zoom_start=12)
    folium.Marker([center_point[0], center_point[1]], popup="center", icon=folium.Icon(color="red")).add_to(map)
    folium.CircleMarker([center_point[0], center_point[1]], radius=100, color="skyblue", fill_color="skyblue").add_to(map)
    for i in range(len(coordinate)):
        folium.Marker([latitude[i], longitude[i]], popup="출발지점 {}, {}km 거리입니다.".format(i, round(haversine(coordinate[i], center_point), 2))).add_to(map)
        folium.PolyLine(locations=[coordinate[i], center_point],weight=2, color = 'blue').add_to(map)
    return map

In [47]:
center_pointer(3)

도로명주소를 입력하세요: 서울특별시 중구 다산로 46길 17
도로명주소를 입력하세요: 서울특별시 강남구 테헤란로 지하340
도로명주소를 입력하세요: 서울특별시 동작구 흑석로 47


서울특별시 중구 다산로 46길 17  
서울특별시 강남구 테헤란로 지하340  
서울특별시 동작구 흑석로 47

In [50]:
review_list = pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/망고플레이트 리뷰 종합.csv", sep=",", index_col=0)
review_list.columns

Index(['이름', '평점', '주소', '주력 메뉴', '가격대', '망고플레이트 주소', '리뷰내용'], dtype='object')

In [59]:
name_address = review_list.drop(["주력 메뉴", '가격대', '망고플레이트 주소', '리뷰내용'], axis=1)
name_address['평점'] = name_address['평점'].str.replace('\n','')

In [65]:
name_address=name_address.drop_duplicates(["이름"])
name_address

이름   평점                        주소
0            청우참치  4.5        서울특별시 강남구 강남대로 632
15             정돈  4.3  서울특별시 강남구 강남대로110길 26 1F
29          꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62
34            노들강  4.4     서울특별시 강남구 강남대로114길 19
41         영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F
...           ...  ...                       ...
36196       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3
36197  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188
36202     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F
36204       조개나루터    -      서울특별시 중랑구 중랑천로12길 59
36206       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61

[3644 rows x 3 columns]

In [73]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

latitude=[]
longitude=[]
coordinate=[]

for i in range(len(name_address)):
    geocode_result = gmaps.geocode(name_address["주소"].iloc[i])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lng=location.get("lng")
    lat=location.get("lat")
    latitude.append(lat)
    longitude.append(lng)
    coordinate.append((lat, lng))

In [72]:
name_address["주소"]

0              서울특별시 강남구 강남대로 632
15       서울특별시 강남구 강남대로110길 26 1F
29          서울특별시 강남구 강남대로110길 62
34          서울특별시 강남구 강남대로114길 19
41       서울특별시 강남구 강남대로124길 10 1F
                   ...           
36196         서울특별시 중랑구 중랑역로53길 3
36197          서울특별시 중랑구 중랑천로 188
36202     서울특별시 중랑구 중랑천로10길 92 1F
36204        서울특별시 중랑구 중랑천로12길 59
36206        서울특별시 중랑구 중랑천로12길 61
Name: 주소, Length: 3644, dtype: object

In [74]:
len(coordinate)

3644

In [75]:
name_address2=name_address.reset_index()
name_address2

index          이름   평점                        주소
0         0        청우참치  4.5        서울특별시 강남구 강남대로 632
1        15          정돈  4.3  서울특별시 강남구 강남대로110길 26 1F
2        29       꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62
3        34         노들강  4.4     서울특별시 강남구 강남대로114길 19
4        41      영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F
...     ...         ...  ...                       ...
3639  36196       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3
3640  36197  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188
3641  36202     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F
3642  36204       조개나루터    -      서울특별시 중랑구 중랑천로12길 59
3643  36206       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61

[3644 rows x 4 columns]

In [78]:
sr_coordinate=pd.Series(coordinate)
sr_coordinate

0        (37.5182046, 127.019356)
1       (37.5035911, 127.0269451)
2       (37.5033264, 127.0294865)
3       (37.5061534, 127.0258181)
4       (37.5088173, 127.0234738)
                  ...            
3639    (37.6163946, 127.0781742)
3640     (37.6012991, 127.074441)
3641     (37.5926809, 127.078032)
3642    (37.5931064, 127.0761017)
3643    (37.5931299, 127.0762339)
Length: 3644, dtype: object

In [79]:
name_coordinate=pd.concat([name_address2, sr_coordinate], axis=1)
name_coordinate

index          이름  ...                        주소                          0
0         0        청우참치  ...        서울특별시 강남구 강남대로 632   (37.5182046, 127.019356)
1        15          정돈  ...  서울특별시 강남구 강남대로110길 26 1F  (37.5035911, 127.0269451)
2        29       꼼다비뛰드  ...     서울특별시 강남구 강남대로110길 62  (37.5033264, 127.0294865)
3        34         노들강  ...     서울특별시 강남구 강남대로114길 19  (37.5061534, 127.0258181)
4        41      영동소금구이  ...  서울특별시 강남구 강남대로124길 10 1F  (37.5088173, 127.0234738)
...     ...         ...  ...                       ...                        ...
3639  36196       커피맬러드  ...       서울특별시 중랑구 중랑역로53길 3  (37.6163946, 127.0781742)
3640  36197  임현숙의이화김치찌개  ...        서울특별시 중랑구 중랑천로 188   (37.6012991, 127.074441)
3641  36202     키미노돼지국밥  ...   서울특별시 중랑구 중랑천로10길 92 1F   (37.5926809, 127.078032)
3642  36204       조개나루터  ...      서울특별시 중랑구 중랑천로12길 59  (37.5931064, 127.0761017)
3643  36206       제주돈선생  ...      서울특별시 중랑구 중랑천로12길 61  (37.5931299, 127.0762339)

[3644 rows x 5 columns]

In [80]:
name_coordinate2=name_coordinate.drop(["index"], axis=1)
name_coordinate2

이름   평점                        주소                          0
0           청우참치  4.5        서울특별시 강남구 강남대로 632   (37.5182046, 127.019356)
1             정돈  4.3  서울특별시 강남구 강남대로110길 26 1F  (37.5035911, 127.0269451)
2          꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62  (37.5033264, 127.0294865)
3            노들강  4.4     서울특별시 강남구 강남대로114길 19  (37.5061534, 127.0258181)
4         영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F  (37.5088173, 127.0234738)
...          ...  ...                       ...                        ...
3639       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3  (37.6163946, 127.0781742)
3640  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188   (37.6012991, 127.074441)
3641     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F   (37.5926809, 127.078032)
3642       조개나루터    -      서울특별시 중랑구 중랑천로12길 59  (37.5931064, 127.0761017)
3643       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61  (37.5931299, 127.0762339)

[3644 rows x 4 columns]

In [84]:
name_coordinate3=name_coordinate2.rename(columns={0:"좌표"})
name_coordinate3

이름   평점                        주소                         좌표
0           청우참치  4.5        서울특별시 강남구 강남대로 632   (37.5182046, 127.019356)
1             정돈  4.3  서울특별시 강남구 강남대로110길 26 1F  (37.5035911, 127.0269451)
2          꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62  (37.5033264, 127.0294865)
3            노들강  4.4     서울특별시 강남구 강남대로114길 19  (37.5061534, 127.0258181)
4         영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F  (37.5088173, 127.0234738)
...          ...  ...                       ...                        ...
3639       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3  (37.6163946, 127.0781742)
3640  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188   (37.6012991, 127.074441)
3641     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F   (37.5926809, 127.078032)
3642       조개나루터    -      서울특별시 중랑구 중랑천로12길 59  (37.5931064, 127.0761017)
3643       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61  (37.5931299, 127.0762339)

[3644 rows x 4 columns]

In [85]:
name_coordinate3.to_csv("/content/drive/MyDrive/맛집 리스트_좌표포함.csv", sep=",")

In [145]:
gu=[]
for i in range(len(name_coordinate2["주소"])):
    gu.append(name_coordinate2["주소"][i][6:9])
#name_coordinate3=pd.concat([name_coordinate3, pd.Series(gu)], axis=1)
pd.Series(gu).unique()

array(['강남구', '강북구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문', '동작구',
       '마포구', '서대문', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포', '용산구',
       '은평구', '종로구', '중구 ', '중랑구'], dtype=object)

In [95]:
name_coordinate3.drop(["구"], axis=1, inplace=True)
name_coordinate3.rename(columns={0:"구"}, inplace=True)
name_coordinate3

이름   평점                        주소                         좌표    구
0           청우참치  4.5        서울특별시 강남구 강남대로 632   (37.5182046, 127.019356)  강남구
1             정돈  4.3  서울특별시 강남구 강남대로110길 26 1F  (37.5035911, 127.0269451)  강남구
2          꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62  (37.5033264, 127.0294865)  강남구
3            노들강  4.4     서울특별시 강남구 강남대로114길 19  (37.5061534, 127.0258181)  강남구
4         영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F  (37.5088173, 127.0234738)  강남구
...          ...  ...                       ...                        ...  ...
3639       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3  (37.6163946, 127.0781742)  중랑구
3640  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188   (37.6012991, 127.074441)  중랑구
3641     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F   (37.5926809, 127.078032)  중랑구
3642       조개나루터    -      서울특별시 중랑구 중랑천로12길 59  (37.5931064, 127.0761017)  중랑구
3643       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61  (37.5931299, 127.0762339)  중랑구

[3644 rows x 5 columns]

In [96]:
name_coordinate3.to_csv("/content/drive/MyDrive/맛집 리스트_좌표포함.csv", sep=",")

In [99]:
name_coordinate3["구"].unique()

array(['강남구', '강북구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문', '동작구',
       '마포구', '서대문', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포', '용산구',
       '은평구', '종로구', '중구 ', '중랑구'], dtype=object)

In [ ]:
#강북구, 금천구, 성북구 삭제 --> 이후 강동구, 강서구, 관악구까지 합쳐서 6개구 추가
idx1=name_coordinate3[name_coordinate3["구"]=="강북구"].index
idx2=name_coordinate3[name_coordinate3["구"]=="금천구"].index
idx3=name_coordinate3[name_coordinate3["구"]=="성북구"].index
idx=[]
for i in idx1:
    idx.append(i)

for i in idx2:
    idx.append(i)

for i in idx3:
    idx.append(i)

idx

In [140]:
name_coordinate4=name_coordinate3.drop(idx)
name_coordinate4

이름   평점                        주소                         좌표    구
0           청우참치  4.5        서울특별시 강남구 강남대로 632   (37.5182046, 127.019356)  강남구
1             정돈  4.3  서울특별시 강남구 강남대로110길 26 1F  (37.5035911, 127.0269451)  강남구
2          꼼다비뛰드  4.5     서울특별시 강남구 강남대로110길 62  (37.5033264, 127.0294865)  강남구
3            노들강  4.4     서울특별시 강남구 강남대로114길 19  (37.5061534, 127.0258181)  강남구
4         영동소금구이  4.4  서울특별시 강남구 강남대로124길 10 1F  (37.5088173, 127.0234738)  강남구
...          ...  ...                       ...                        ...  ...
3639       커피맬러드    -       서울특별시 중랑구 중랑역로53길 3  (37.6163946, 127.0781742)  중랑구
3640  임현숙의이화김치찌개    -        서울특별시 중랑구 중랑천로 188   (37.6012991, 127.074441)  중랑구
3641     키미노돼지국밥    -   서울특별시 중랑구 중랑천로10길 92 1F   (37.5926809, 127.078032)  중랑구
3642       조개나루터    -      서울특별시 중랑구 중랑천로12길 59  (37.5931064, 127.0761017)  중랑구
3643       제주돈선생    -      서울특별시 중랑구 중랑천로12길 61  (37.5931299, 127.0762339)  중랑구

[3491 rows x 5 columns]

In [142]:
for gu in name_coordinate4["구"].unique():
    print(gu, " : ", len(name_coordinate4[name_coordinate4["구"]==gu]))

강남구  :  198
광진구  :  195
구로구  :  181
노원구  :  172
도봉구  :  184
동대문  :  191
동작구  :  188
마포구  :  190
서대문  :  190
서초구  :  186
성동구  :  189
송파구  :  187
양천구  :  171
영등포  :  178
용산구  :  178
은평구  :  179
종로구  :  179
중구   :  181
중랑구  :  174


In [143]:
name_coordinate4.to_csv("/content/drive/MyDrive/맛집 리스트_강북금천성북구 삭제.csv", sep=",")

In [111]:
aa = review_list["주소"].unique()
gu=[]
for i in range(len(aa)):
    gu.append(aa[i][6:9])

pd.Series(gu).unique()

array(['강남구', '강북구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문', '동작구',
       '마포구', '서대문', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포', '용산구',
       '은평구', '종로구', '중구 ', '중랑구'], dtype=object)

In [ ]:
강남구 강동구 강북구 강서구 관악구 광진구 구로구 금천구 노원구 도봉구 동대문구 동작구 마포구 서대문구 서초구 성동구 성북구 송파구 양천구 영등포구 용산구 은평구 종로구 중구 중랑구

In [113]:
gd = pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/강동구 리뷰.csv", sep=",", index_col=0)
bsa = pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/강북, 강서, 관악구 리뷰.csv", sep=",", index_col=0)

In [114]:
gd2=gd.drop_duplicates(["주소"])
gd2

이름  ...                                               리뷰내용
0        얌얌카츠  ...  \n                    사진 더보기\n                ...
58        보타이  ...  \n                    사진 더보기\n                ...
116    이한진숙성회  ...  \n                    사진 더보기\n                ...
174      경양미식  ...  \n                    사진 더보기\n                ...
226    프릭스다이너  ...  \n                    사진 더보기\n                ...
...       ...  ...                                                ...
9091  길위의 파스타  ...  \n                    사진 더보기\n                ...
9133     고기감성  ...  \n                    사진 더보기\n                ...
9174     스타벅스  ...  \n                    사진 더보기\n                ...
9215     한신포차  ...  \n                    사진 더보기\n                ...
9254  한왕스1980  ...  \n                    사진 더보기\n                ...

[196 rows x 7 columns]

In [119]:
bsa2=bsa.drop_duplicates(["주소"])
bsa2

이름  ...                                               리뷰내용
0        양다리걸쳤네  ...  \n                    사진 더보기\n                ...
52       장수마늘보쌈  ...  \n                    사진 더보기\n                ...
113        삼덕식당  ...  \n                    사진 더보기\n                ...
165        동네함박  ...  \n                    사진 더보기\n                ...
227      다정도병인양  ...  \n                    사진 더보기\n                ...
...         ...  ...                                                ...
27797       길초밥  ...  \n                    사진 더보기\n                ...
27847  생각보다맛있는집  ...  \n                    사진 더보기\n                ...
27904       노다지  ...  \n                    사진 더보기\n                ...
27958       황토방  ...  \n                    사진 더보기\n                ...
28009     자하연식당  ...  \n                    사진 더보기\n                ...

[587 rows x 7 columns]

In [123]:
sb=pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/성북구 리뷰.csv", sep=",", index_col=0)
sb2=sb.drop_duplicates(["주소"])
sb2

이름  ...                                               리뷰내용
0        우정초밥  ...  \n                    사진 더보기\n                ...
62         카레  ...  \n                    사진 더보기\n                ...
124        공푸  ...  \n                    사진 더보기\n                ...
186        모짜  ...  \n                    사진 더보기\n                ...
245    임금님자장면  ...  \n                    사진 더보기\n                ...
...       ...  ...                                                ...
10249    동병상련  ...  \n                    사진 더보기\n                ...
10296     카페잇  ...  \n                    사진 더보기\n                ...
10335    숙수도가  ...  \n                    사진 더보기\n                ...
10374  프로젝트얼스  ...  \n                    사진 더보기\n                ...
10428  Red불닭발  ...  \n                    사진 더보기\n                ...

[195 rows x 7 columns]

In [131]:
gc=pd.read_csv("/content/drive/MyDrive/에이콘/파이널 프로젝트/data/금천구 리뷰.csv", sep=",", index_col=0)
gc2=gc.drop_duplicates(["주소"])
gc2

이름  ...                                               리뷰내용
0             여신족발  ...  \n                    사진 더보기\n                ...
59    김사부샤브샤브 무한리필  ...  \n                    사진 더보기\n                ...
126           진영면옥  ...  \n                    사진 더보기\n                ...
167         세라즈피자펍  ...  \n                    사진 더보기\n                ...
219          카페스미다  ...  \n                    사진 더보기\n                ...
...            ...  ...                                                ...
7464      막회무침나라횟집  ...  \n                    사진 더보기\n                ...
7499        포차인닭갈비  ...  \n                    사진 더보기\n                ...
7537          명화곱창  ...  \n                    사진 더보기\n                ...
7574        혜화동돈까스  ...  \n                    사진 더보기\n                ...
7608          에그스탑  ...  \n                    사진 더보기\n                ...

[185 rows x 7 columns]

In [132]:
aa= pd.concat([gd2, sb2])
aa= pd.concat([aa, bsa2])
aa= pd.concat([aa, gc2])
aa

이름  ...                                               리뷰내용
0         얌얌카츠  ...  \n                    사진 더보기\n                ...
58         보타이  ...  \n                    사진 더보기\n                ...
116     이한진숙성회  ...  \n                    사진 더보기\n                ...
174       경양미식  ...  \n                    사진 더보기\n                ...
226     프릭스다이너  ...  \n                    사진 더보기\n                ...
...        ...  ...                                                ...
7464  막회무침나라횟집  ...  \n                    사진 더보기\n                ...
7499    포차인닭갈비  ...  \n                    사진 더보기\n                ...
7537      명화곱창  ...  \n                    사진 더보기\n                ...
7574    혜화동돈까스  ...  \n                    사진 더보기\n                ...
7608      에그스탑  ...  \n                    사진 더보기\n                ...

[1163 rows x 7 columns]

In [134]:
aa2=aa.drop(["주력 메뉴", "가격대", "망고플레이트 주소", "리뷰내용"], axis=1)
aa2

이름       평점                                    주소
0         얌얌카츠  \n4.4\n                   서울특별시 강동구 천호옛14길 22
58         보타이  \n4.3\n              서울특별시 강동구 풍성로 90 한스빌딩 1F
116     이한진숙성회  \n4.3\n                  서울특별시 강동구 구천면로29길 64
174       경양미식  \n4.3\n                 서울특별시 강동구 천호옛14길 23-4
226     프릭스다이너  \n4.3\n                 서울특별시 강동구 천호대로159길 53
...        ...      ...                                   ...
7464  막회무침나라횟집  최신 버전 앱                  서울특별시 금천구 시흥대로71길 40
7499    포차인닭갈비  최신 버전 앱  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하
7537      명화곱창  최신 버전 앱                    서울특별시 금천구 벚꽃로38길 5
7574    혜화동돈까스  최신 버전 앱                  서울특별시 금천구 금하로5길 4 1F
7608      에그스탑  최신 버전 앱      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F

[1163 rows x 3 columns]

In [147]:
aa2['평점'] = aa2['평점'].str.replace('\n','')
aa2=aa2.reset_index()

In [148]:
gu=[]
for i in range(len(aa2["주소"])):
    gu.append(aa2["주소"][i][6:9])
aa3=pd.concat([aa2, pd.Series(gu)], axis=1)
aa3

index        이름       평점                                    주소    0
0         0      얌얌카츠      4.4                   서울특별시 강동구 천호옛14길 22  강동구
1        58       보타이      4.3              서울특별시 강동구 풍성로 90 한스빌딩 1F  강동구
2       116    이한진숙성회      4.3                  서울특별시 강동구 구천면로29길 64  강동구
3       174      경양미식      4.3                 서울특별시 강동구 천호옛14길 23-4  강동구
4       226    프릭스다이너      4.3                 서울특별시 강동구 천호대로159길 53  강동구
...     ...       ...      ...                                   ...  ...
1158   7464  막회무침나라횟집  최신 버전 앱                  서울특별시 금천구 시흥대로71길 40  금천구
1159   7499    포차인닭갈비  최신 버전 앱  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하  금천구
1160   7537      명화곱창  최신 버전 앱                    서울특별시 금천구 벚꽃로38길 5  금천구
1161   7574    혜화동돈까스  최신 버전 앱                  서울특별시 금천구 금하로5길 4 1F  금천구
1162   7608      에그스탑  최신 버전 앱      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F  금천구

[1163 rows x 5 columns]

In [149]:
aa3.drop(["index"], axis=1, inplace=True)
aa3

이름       평점                                    주소    0
0         얌얌카츠      4.4                   서울특별시 강동구 천호옛14길 22  강동구
1          보타이      4.3              서울특별시 강동구 풍성로 90 한스빌딩 1F  강동구
2       이한진숙성회      4.3                  서울특별시 강동구 구천면로29길 64  강동구
3         경양미식      4.3                 서울특별시 강동구 천호옛14길 23-4  강동구
4       프릭스다이너      4.3                 서울특별시 강동구 천호대로159길 53  강동구
...        ...      ...                                   ...  ...
1158  막회무침나라횟집  최신 버전 앱                  서울특별시 금천구 시흥대로71길 40  금천구
1159    포차인닭갈비  최신 버전 앱  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하  금천구
1160      명화곱창  최신 버전 앱                    서울특별시 금천구 벚꽃로38길 5  금천구
1161    혜화동돈까스  최신 버전 앱                  서울특별시 금천구 금하로5길 4 1F  금천구
1162      에그스탑  최신 버전 앱      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F  금천구

[1163 rows x 4 columns]

In [151]:
aa3.to_excel("/content/drive/MyDrive/빠진 구 정리.xlsx")
aa3.to_csv("/content/drive/MyDrive/빠진 구 정리.csv", sep=",")

In [157]:
missing=pd.read_excel("/content/drive/MyDrive/에이콘/0106 등원작업물/빠진구정리.xlsx", index_col=0)
missing.reset_index(inplace=True)
missing.drop(["index"], axis=1, inplace=True)

In [158]:
gmaps = googlemaps.Client(key='AIzaSyCNetZW4eZiBmY7Zlwa9IDOj1CtZhgm_a0')

latitude=[]
longitude=[]
coordinate=[]

for i in range(len(missing)):
    geocode_result = gmaps.geocode(missing["주소"].iloc[i])
    geometry=geocode_result[0].get("geometry")
    location=geometry.get("location")
    lng=location.get("lng")
    lat=location.get("lat")
    latitude.append(lat)
    longitude.append(lng)
    coordinate.append((lat, lng))

In [159]:
sr_coordinate=pd.Series(coordinate)
sr_coordinate

0              (37.5359894, 127.1266811)
1        (37.55195610000001, 127.134788)
2              (37.5425823, 127.1310616)
3              (37.5362293, 127.1269566)
4              (37.5398434, 127.1281785)
                      ...               
1158    (37.45521069999999, 126.8993725)
1159           (37.4800992, 126.8826202)
1160           (37.4818785, 126.8832991)
1161           (37.4543357, 126.8988835)
1162           (37.4806318, 126.8805702)
Length: 1163, dtype: object

In [163]:
missing_coordinate=pd.concat([missing, sr_coordinate], axis=1)
missing_coordinate.rename(columns={0:"좌표"}, inplace=True)
missing_coordinate

이름   평점  ...    구                                좌표
0         얌얌카츠  4.4  ...  강동구         (37.5359894, 127.1266811)
1          보타이  4.3  ...  강동구   (37.55195610000001, 127.134788)
2       이한진숙성회  4.3  ...  강동구         (37.5425823, 127.1310616)
3         경양미식  4.3  ...  강동구         (37.5362293, 127.1269566)
4       프릭스다이너  4.3  ...  강동구         (37.5398434, 127.1281785)
...        ...  ...  ...  ...                               ...
1158  막회무침나라횟집    -  ...  금천구  (37.45521069999999, 126.8993725)
1159    포차인닭갈비    -  ...  금천구         (37.4800992, 126.8826202)
1160      명화곱창    -  ...  금천구         (37.4818785, 126.8832991)
1161    혜화동돈까스    -  ...  금천구         (37.4543357, 126.8988835)
1162      에그스탑    -  ...  금천구         (37.4806318, 126.8805702)

[1163 rows x 5 columns]

In [164]:
name_complete=pd.concat([name_coordinate4, missing_coordinate], axis=0)
name_complete

이름   평점  ...                                좌표    구
0         청우참치  4.5  ...          (37.5182046, 127.019356)  강남구
1           정돈  4.3  ...         (37.5035911, 127.0269451)  강남구
2        꼼다비뛰드  4.5  ...         (37.5033264, 127.0294865)  강남구
3          노들강  4.4  ...         (37.5061534, 127.0258181)  강남구
4       영동소금구이  4.4  ...         (37.5088173, 127.0234738)  강남구
...        ...  ...  ...                               ...  ...
1158  막회무침나라횟집    -  ...  (37.45521069999999, 126.8993725)  금천구
1159    포차인닭갈비    -  ...         (37.4800992, 126.8826202)  금천구
1160      명화곱창    -  ...         (37.4818785, 126.8832991)  금천구
1161    혜화동돈까스    -  ...         (37.4543357, 126.8988835)  금천구
1162      에그스탑    -  ...         (37.4806318, 126.8805702)  금천구

[4654 rows x 5 columns]

In [165]:
for i in name_complete["구"].unique():
    print(i, " : ", len(name_complete[name_complete["구"]==i]))

강남구  :  198
광진구  :  196
구로구  :  181
노원구  :  172
도봉구  :  184
동대문  :  191
동작구  :  188
마포구  :  190
서대문  :  190
서초구  :  186
성동구  :  189
송파구  :  187
양천구  :  171
영등포  :  178
용산구  :  178
은평구  :  179
종로구  :  179
중구   :  181
중랑구  :  174
강동구  :  195
성북구  :  195
강북구  :  198
강서구  :  193
관악구  :  196
금천구  :  185


In [167]:
name_complete.to_csv("/content/drive/MyDrive/좌표포함 맛집리스트 완성본.csv", sep=",")

In [ ]:
## 앞으로 해야할일 --> 구이름 고치기, 오늘 메꾼 6개구 리뷰까지 다시 합치기

In [6]:
name_complete=pd.read_csv("C:\jupyter_notebook\에이콘강의\파이널 프로젝트\data\좌표포함 맛집리스트.csv", sep=",", index_col=0)
name_complete.reset_index(inplace=True)
name_complete.drop(["index"], axis=1, inplace=True)
name_complete

이름   평점                                    주소  \
0         청우참치  4.5                    서울특별시 강남구 강남대로 632   
1           정돈  4.3              서울특별시 강남구 강남대로110길 26 1F   
2        꼼다비뛰드  4.5                 서울특별시 강남구 강남대로110길 62   
3          노들강  4.4                 서울특별시 강남구 강남대로114길 19   
4       영동소금구이  4.4              서울특별시 강남구 강남대로124길 10 1F   
...        ...  ...                                   ...   
4649  막회무침나라횟집    -                  서울특별시 금천구 시흥대로71길 40   
4650    포차인닭갈비    -  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하   
4651      명화곱창    -                    서울특별시 금천구 벚꽃로38길 5   
4652    혜화동돈까스    -                  서울특별시 금천구 금하로5길 4 1F   
4653      에그스탑    -      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F   

                                    좌표    구  
0             (37.5182046, 127.019356)  강남구  
1            (37.5035911, 127.0269451)  강남구  
2            (37.5033264, 127.0294865)  강남구  
3            (37.5061534, 127.0258181)  강남구  
4            (37.5088173, 127.0234738)  강남구  
...                                ...  ...  
4649  (37.45521069999999, 126.8993725)  금천구  
4650         (37.4800992, 126.8826202)  금천구  
4651         (37.4818785, 126.8832991)  금천구  
4652         (37.4543357, 126.8988835)  금천구  
4653         (37.4806318, 126.8805702)  금천구  

[4654 rows x 5 columns]

In [9]:
name_complete["구"].unique()
#len(name_complete["구"].unique())

array(['강남구', '광진구', '구로구', '노원구', '도봉구', '동대문', '동작구', '마포구', '서대문',
       '서초구', '성동구', '송파구', '양천구', '영등포', '용산구', '은평구', '종로구', '중구 ',
       '중랑구', '강동구', '성북구', '강북구', '강서구', '관악구', '금천구'], dtype=object)

In [10]:
def guname(x):
    if x=="동대문":
        return "동대문구"
    elif x=="서대문":
        return "서대문구"
    elif x=="영등포":
        return "영등포구"
    elif x=="중구 ":
        return "중구"
    else:
        return x

In [13]:
name_complete["구"]=name_complete["구"].apply(guname)
name_complete["구"].unique()

array(['강남구', '광진구', '구로구', '노원구', '도봉구', '동대문구', '동작구', '마포구', '서대문구',
       '서초구', '성동구', '송파구', '양천구', '영등포구', '용산구', '은평구', '종로구', '중구',
       '중랑구', '강동구', '성북구', '강북구', '강서구', '관악구', '금천구'], dtype=object)

In [15]:
name_complete.to_csv("C:\jupyter_notebook\에이콘강의\파이널 프로젝트\data\맛집리스트_구이름수정.csv", sep=",")

In [22]:

name_complete['좌표2'] = name_complete['좌표'].str.replace('(','')
name_complete['좌표2'] = name_complete['좌표2'].str.replace(')','')
name_complete

이름   평점                                    주소  \
0         청우참치  4.5                    서울특별시 강남구 강남대로 632   
1           정돈  4.3              서울특별시 강남구 강남대로110길 26 1F   
2        꼼다비뛰드  4.5                 서울특별시 강남구 강남대로110길 62   
3          노들강  4.4                 서울특별시 강남구 강남대로114길 19   
4       영동소금구이  4.4              서울특별시 강남구 강남대로124길 10 1F   
...        ...  ...                                   ...   
4649  막회무침나라횟집    -                  서울특별시 금천구 시흥대로71길 40   
4650    포차인닭갈비    -  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하   
4651      명화곱창    -                    서울특별시 금천구 벚꽃로38길 5   
4652    혜화동돈까스    -                  서울특별시 금천구 금하로5길 4 1F   
4653      에그스탑    -      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F   

                                    좌표    구                             좌표2  
0             (37.5182046, 127.019356)  강남구          37.5182046, 127.019356  
1            (37.5035911, 127.0269451)  강남구         37.5035911, 127.0269451  
2            (37.5033264, 127.0294865)  강남구         37.5033264, 127.0294865  
3            (37.5061534, 127.0258181)  강남구         37.5061534, 127.0258181  
4            (37.5088173, 127.0234738)  강남구         37.5088173, 127.0234738  
...                                ...  ...                             ...  
4649  (37.45521069999999, 126.8993725)  금천구  37.45521069999999, 126.8993725  
4650         (37.4800992, 126.8826202)  금천구         37.4800992, 126.8826202  
4651         (37.4818785, 126.8832991)  금천구         37.4818785, 126.8832991  
4652         (37.4543357, 126.8988835)  금천구         37.4543357, 126.8988835  
4653         (37.4806318, 126.8805702)  금천구         37.4806318, 126.8805702  

[4654 rows x 6 columns]

In [23]:
name_complete["위도"]=name_complete["좌표2"].str.split(", ").str[0]
name_complete["경도"]=name_complete["좌표2"].str.split(", ").str[1]
name_complete

이름   평점                                    주소  \
0         청우참치  4.5                    서울특별시 강남구 강남대로 632   
1           정돈  4.3              서울특별시 강남구 강남대로110길 26 1F   
2        꼼다비뛰드  4.5                 서울특별시 강남구 강남대로110길 62   
3          노들강  4.4                 서울특별시 강남구 강남대로114길 19   
4       영동소금구이  4.4              서울특별시 강남구 강남대로124길 10 1F   
...        ...  ...                                   ...   
4649  막회무침나라횟집    -                  서울특별시 금천구 시흥대로71길 40   
4650    포차인닭갈비    -  서울특별시 금천구 가산디지털1로 168 우림라이온스밸리 A동 지하   
4651      명화곱창    -                    서울특별시 금천구 벚꽃로38길 5   
4652    혜화동돈까스    -                  서울특별시 금천구 금하로5길 4 1F   
4653      에그스탑    -      서울특별시 금천구 가산디지털1로 171 SK V1빌딩 1F   

                                    좌표    구                             좌표2  \
0             (37.5182046, 127.019356)  강남구          37.5182046, 127.019356   
1            (37.5035911, 127.0269451)  강남구         37.5035911, 127.0269451   
2            (37.5033264, 127.0294865)  강남구         37.5033264, 127.0294865   
3            (37.5061534, 127.0258181)  강남구         37.5061534, 127.0258181   
4            (37.5088173, 127.0234738)  강남구         37.5088173, 127.0234738   
...                                ...  ...                             ...   
4649  (37.45521069999999, 126.8993725)  금천구  37.45521069999999, 126.8993725   
4650         (37.4800992, 126.8826202)  금천구         37.4800992, 126.8826202   
4651         (37.4818785, 126.8832991)  금천구         37.4818785, 126.8832991   
4652         (37.4543357, 126.8988835)  금천구         37.4543357, 126.8988835   
4653         (37.4806318, 126.8805702)  금천구         37.4806318, 126.8805702   

                     위도           경도  
0            37.5182046   127.019356  
1            37.5035911  127.0269451  
2            37.5033264  127.0294865  
3            37.5061534  127.0258181  
4            37.5088173  127.0234738  
...                 ...          ...  
4649  37.45521069999999  126.8993725  
4650         37.4800992  126.8826202  
4651         37.4818785  126.8832991  
4652         37.4543357  126.8988835  
4653         37.4806318  126.8805702  

[4654 rows x 8 columns]

In [24]:
name_complete.to_csv("C:\jupyter_notebook\에이콘강의\파이널 프로젝트\data\맛집리스트_위도경도분리(0108).csv", sep=",")

In [19]:

map=folium.Map(location=[37.335887, 126.584063], zoom_start=12)
#folium.Marker([center_point[0], center_point[1]], popup="center", icon=folium.Icon(color="red")).add_to(map)
#folium.CircleMarker([center_point[0], center_point[1]], radius=100, color="skyblue", fill_color="skyblue").add_to(map)
for i in range(len(gng)):
    folium.Marker([gng["좌표"]], popup="{}".format(name_complete[name_complete["좌표"]==gng[i]["좌표"]]["이름"])).add_to(map)


이름   평점                            주소  \
0           청우참치  4.5            서울특별시 강남구 강남대로 632   
1             정돈  4.3      서울특별시 강남구 강남대로110길 26 1F   
2          꼼다비뛰드  4.5         서울특별시 강남구 강남대로110길 62   
3            노들강  4.4         서울특별시 강남구 강남대로114길 19   
4         영동소금구이  4.4      서울특별시 강남구 강남대로124길 10 1F   
..           ...  ...                           ...   
193          리북집  4.3            서울특별시 강남구 학동로2길 47   
194          수묵당  4.4           서울특별시 강남구 학동로53길 20   
195          무오키  4.3        서울특별시 강남구 학동로55길 12-12   
196  리틀앤머치 (휴업중)  4.4   서울특별시 강남구 학동로56길 49 성림빌딩 1F   
197           톡톡  4.5  서울특별시 강남구 학동로97길 41  리유빌딩 3F   

                                   좌표    구  
0            (37.5182046, 127.019356)  강남구  
1           (37.5035911, 127.0269451)  강남구  
2           (37.5033264, 127.0294865)  강남구  
3           (37.5061534, 127.0258181)  강남구  
4           (37.5088173, 127.0234738)  강남구  
..                                ...  ...  
193  (37.50858729999999, 127.0236601)  강남구  
194         (37.5184075, 127.0399926)  강남구  
195         (37.5189698, 127.0422219)  강남구  
196         (37.5139868, 127.0435202)  강남구  
197  (37.52237119999999, 127.0542979)  강남구  

[198 rows x 5 columns]